File: Difficulty: Medium/Minimum Vertical Sum/minimum-vertical-sum.java
//{ Driver Code Starts
// Initial Template for Java

/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;
import java.util.ListIterator;

class GFG {
    public static void main(String[] args) {
        // Creating an object of class Scanner
        Scanner sc = new Scanner(System.in);

        // Taking input the total number of testcases
        int t = sc.nextInt();

        while (t-- > 0) {
            // Taking the total number of rows
            int n = sc.nextInt();
            sc.nextLine();
            // Declaring and Initalizing an ArrayList
            ArrayList<ArrayList<Integer>> arr = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                // Taking the number of columns
                String tempS[] = sc.nextLine().trim().split(" ");
                // Declaring and Intializing an ArrayList for each row
                ArrayList<Integer> temp = new ArrayList<>();
                int m = tempS.length;
                for (int j = 0; j < m; j++) {
                    temp.add(Integer.parseInt(tempS[j]));
                }

                arr.add(temp);
            }

            // Calling the method minimum_verical_sum
            System.out.println(minimumVerticalSum(arr));
        }
    }

    
// } Driver Code Ends

// User function Template for Java

public static int minimumVerticalSum(ArrayList<ArrayList<Integer>> arr) {
    // Initialize variable to track the maximum number of columns
    int maxLength = 0;
    
    // Find the maximum length of any column (list)
    for (ArrayList<Integer> column : arr) {
        maxLength = Math.max(maxLength, column.size());
    }
    
    // Array to store the sum of each column
    int[] columnSum = new int[maxLength];
    
    // Calculate the sum of elements for each column
    for (ArrayList<Integer> column : arr) {
        for (int j = 0; j < column.size(); j++) {
            columnSum[j] += column.get(j);
        }
    }
    
    // Find the minimum vertical sum from the calculated column sums
    int minSum = Integer.MAX_VALUE;
    for (int sum : columnSum) {
        minSum = Math.min(minSum, sum);
    }
    
    // Return the minimum vertical sum
    return minSum;
}


//{ Driver Code Starts.
}
// } Driver Code Ends

File: Difficulty: Medium/Minimum Vertical Sum/minimum-vertical-sum.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    int minimum_vertical_sum(vector<vector<int>>& arr, int n) {
        /**
         * Function to find the minimum vertical sum in a 2D matrix.
         * 
         * @param arr 2D vector representing the matrix.
         * @param n Number of rows in the matrix.
         * @return Minimum positive vertical sum across all columns.
         */
        
        int minVerticalSum = INT_MAX; // Initialize the minimum sum with a large value

        // Iterate over possible column indices (assuming a max column index of 3005)
        for (int col = 0; col < 3005; col++) {
            int columnSum = 0;

            // Traverse each row and accumulate the sum of the current column
            for (int row = 0; row < n; row++) {
                if (arr[row].size() > col) { // Ensure the column exists in the current row
                    columnSum += arr[row][col];
                }
            }

            // Update the minimum vertical sum if a positive sum is found
            if (columnSum > 0) {
                minVerticalSum = min(columnSum, minVerticalSum);
            }
        }

        return minVerticalSum;
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        cin.ignore();
        vector<vector<int>> arr;
        for (int i = 0; i < n; i++) {
            string line;
            getline(cin, line);
            stringstream ss(line);
            vector<int> arr2;
            int num;
            while (ss >> num) {
                arr2.push_back(num);
            }
            arr.push_back(arr2);
        }
        Solution ob;
        cout << ob.minimum_vertical_sum(arr, n) << endl;
        cout << "~\n";
    }
    return 0;
}

// } Driver Code Ends